### PR TITLE
[docs] Pass router instead of separate url and asPath properties

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Part of [ENG-3964](https://linear.app/expo/issue/ENG-3964/split-up-documentationpagetsx)

We used to pass the `url` and `asPath` all the way in this chain of files:
- **components/page-higher-order/DocumentationElements.tsx**
  - **components/DocumentationPage.tsx**
    - **components/DocumentationFooter.tsx**
    - **components/DocumentationSidebar.tsx**
      - **components/DocumentationSidebarGroup.tsx**
      - **components/DocumentationSidebarLink.tsx**

Preferably, we would just use `useRouter` in most scenarios. In this PR I only replaced both `url` and `asPath` with `router` because we are doing the same thing over and over in each component: detect if the link that's being rendered should be rendered as active.

We should find a better way for this to do this once, and reuse it instead of recalculating it in every component.

# How

- Replaced `url={router}` and `asPath={router.asPath}` with just `router={router}` and simplified the props in the whole chain.

# Test Plan

- Open any page through links
- Check if the link is still marked as active

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by Simek — APPROVED** [review]
LGTM, that's a nice code clarity improvement! 👍

File: docs/components/DocumentationSidebarLink.tsx [hunk 1]
```diff
@@ -77,23 +77,23 @@ export default class DocumentationSidebarLink extends React.Component<Props> {
   }
 
   isSelected() {
-    if (!this.props.url) {
+    if (!this.props.router) {
       console.log('[debug] isSelected bailed out, no url', this.props);
```
Comment (Simek) [inline docs/components/DocumentationSidebarLink.tsx:81]:
```suggestion
      console.log('[debug] isSelected bailed out, no router', this.props);
```
Just spotted this right now, this suggestion is not merge blocking in any way. 🙂

Comment (byCedric) [reply docs/components/DocumentationSidebarLink.tsx:81]: ooh nice, thanks!

Comment (byCedric) [reply docs/components/DocumentationSidebarLink.tsx]: I don't think I have ever seen this log statement though 🙈  There is always a pathname, even if its `https://docs.expo.dev` -> `/`

Comment (Simek) [reply docs/components/DocumentationSidebarLink.tsx]: Maybe we can just remove this log then?

**Pull request merged**